src/Configuration/Provisioning/data/hooks.js

```diff
@@ -0,0 +1,215 @@
+import { useCallback } from 'react';
+import { useContextSelector } from 'use-context-selector';
+import { ProvisioningContext } from '../ProvisioningContext';
+
+export default function useProvisioningContext() {
+  const setState = useContextSelector(ProvisioningContext, (v) => v[1]);
+  const setMultipleFunds = useCallback((fundingBoolean) => {
+    setState(s => ({
+      ...s,
+      multipleFunds: fundingBoolean,
+      formData: {
+        ...s.formData,
+        multipleFunds: fundingBoolean,
+      },
+    }));
+  }, [setState]);
+
+  const setCustomCatalog = useCallback((customCatalogBoolean) => {
+    setState(s => ({
+      ...s,
+      customCatalog: customCatalogBoolean,
+    }));
+  }, [setState]);
+
+  const instatiateMultipleFormData = useCallback((catalogQueryTitle) => {
+    setState(s => {
+      const { formData } = s;
+      return {
+        ...s,
+        formData: {
+          ...formData,
+          policies: catalogQueryTitle,
+        },
+      };
+    });
+  }, [setState]);
+
+  const resetPolicies = useCallback(() => {
+    setState(s => ({
+      ...s,
+      formData: {
+        ...s.formData,
+        policies: [],
+      },
+    }));
+  }, [setState]);
+
+  const setCustomerUUID = useCallback((customerUUID) => {
+    setState(s => ({
+      ...s,
+      formData: {
+        ...s.formData,
+        enterpriseUUID: customerUUID,
+      },
+    }));
+  }, [setState]);
+
+  const setFinancialIdentifier = useCallback((financialIdentifier) => {
+    setState(s => ({
+      ...s,
+      formData: {
+        ...s.formData,
+        financialIdentifier,
+      },
+    }));
+  }, [setState]);
+
+  const setStartDate = useCallback((startDate) => {
+    setState(s => ({
+      ...s,
+      formData: {
+        ...s.formData,
+        startDate,
+      },
+    }));
+  }, [setState]);
+
+  const setEndDate = useCallback((endDate) => {
+    setState(s => ({
+      ...s,
+      formData: {
+        ...s.formData,
+        endDate,
+      },
+    }));
+  }, [setState]);
+
+  const setSubsidyRevReq = useCallback((subsidyRevReq) => {
+    setState(s => ({
+      ...s,
+      formData: {
+        ...s.formData,
+        subsidyRevReq,
+      },
+    }));
+  }, [setState]);
+
+  const setAccountName = useCallback((accountName, index) => {
+    setState(s => {
+      const { policies } = s.formData;
+      policies[index] = {
+        ...policies[index],
+        ...accountName,
+      };
+      const newPolicies = policies.map((policy) => policy);
```

Comment:
If we're just looking to create an array copy
```suggestion
      const newPolicies = [...policies];
```